common/djangoapps/util/milestones_helpers.py

```diff
@@ -0,0 +1,136 @@
+        pre_requisite_course_key = course_milestones[0]['namespace']
+    return pre_requisite_course_key
+
+
+def add_prerequisite_course(course_key, prerequisite_course_key):
+    """
+    adds pre-requisite course milestone to a course
+    """
+    # create a milestone
+    milestone = add_milestone({
+        'name': 'Course {} requires {}'.format(unicode(course_key), unicode(prerequisite_course_key)),
+        'namespace': unicode(prerequisite_course_key),
+        'description': '',
+    })
+    # add requirement course milestone
+    add_course_milestone(course_key, 'requires', milestone)
+
+    # add fulfillment course milestone
+    add_course_milestone(prerequisite_course_key, 'fulfills', milestone)
+
+
+def remove_prerequisite_course(course_key, milestone):
+    """
+    remove pre-requisite course milestone for a course
+    """
+
+    remove_course_milestone(
+        course_key,
+        milestone,
+    )
+
+
+def set_prerequisite_course(course_key, prerequisite_course_key_string):
+    """
+    add or update pre-requisite course milestone for a course
+    """
+    #remove any existing requirement milestones with this pre-requisite course as requirement
+    course_milestones = get_course_milestones(course_key=course_key, relationship="requires")
+    if course_milestones:
+        for milestone in course_milestones:
+            remove_prerequisite_course(course_key, milestone)
+
+    # add milestones if pre-requisite course is selected
+    if prerequisite_course_key_string:
+        prerequisite_course_key = CourseKey.from_string(prerequisite_course_key_string)
+        add_prerequisite_course(course_key, prerequisite_course_key)
+
+
+def get_pre_requisite_courses(course_ids):
+    """
+    It would fetch pre-requisite courses for a list of courses
+
+    Returns a dict with keys are set to course id and values are set to pre-requisite course keys .i.e.
+    {
+        "org/DemoX/2014_T2": {"milestone_id": "1", "prc_id": "org/cs23/2014_T2"}
+    }
+    """
+    courses_dict = defaultdict(dict)
+    milestones = get_courses_milestones(course_ids, relationship="requires")
+    for milestone in milestones:
+        pre_requisite_course_id = milestone['namespace']
+        #TODO check if it is valid course key set as namespace
+        if pre_requisite_course_id:
+            pre_requisite_course_key = CourseKey.from_string(pre_requisite_course_id)
+            course_id = CourseKey.from_string(milestone['course_id'])
+            courses_dict[course_id] = {
+                "milestone_id": milestone['id'],
+                "prc_id": pre_requisite_course_key
+            }
+    return courses_dict
+
+
+def milestones_achieved_by_user(user):
+    """
+    It would fetch list of milestones completed by user
+    """
+    return get_user_milestones(user)
+
+
+def get_prc_not_completed(user, enrolled_courses):
+    """
+    It would fetch list of prerequisite courses not completed by user
```

Comment:
Looks like it actually returns a dict. And the code itself says it will only return a single pre-requisite course.